livelog.go

```diff
@@ -153,30 +157,34 @@ func (l *LiveLogTask) reinstateBackingLog() {
 	}
 }
 
-func (l *LiveLogTask) uploadLiveLog() error {
-	// add an extra 15 minutes, to adequately cover client/server clock drift or task initialisation delays
-	maxRunTimeDeadline := time.Now().Add(time.Duration(l.task.Payload.MaxRunTime+900) * time.Second)
-	// deduce stateless DNS name to use
-	statelessHostname := hostname.New(config.PublicIP, config.Subdomain, maxRunTimeDeadline, config.LiveLogSecret)
-	getURL, err := url.Parse(l.liveLog.GetURL)
+func (l *LiveLogTask) uploadLiveLogArtifact() error {
+	var err error
+	l.exposure, err = exposer.ExposeHTTP(config.LiveLogGETPort)
+	if err != nil {
+		return err
+	}
+
+	// combine the path from the livelog URL with the expose URL
+	logURL, err := url.Parse(l.liveLog.GetURL)
 	if err != nil {
 		return err
 	}
-	if l.liveLog.SSLCert != "" && l.liveLog.SSLKey != "" {
-		getURL.Scheme = "https"
+	exposeURL := l.exposure.GetURL()
+	if exposeURL.Path == "/" {
+		exposeURL.Path = logURL.Path
 	} else {
-		getURL.Scheme = "http"
+		exposeURL.Path = exposeURL.Path + logURL.Path
 	}
-	getURL.Host = statelessHostname + ":" + strconv.Itoa(int(l.liveLog.GETPort))
+
+	expires := time.Now().Add(time.Duration(l.task.Payload.MaxRunTime+900) * time.Second)
```

Comment:
a comment to explain that the 900 is to account for clock drift would be useful here